### PR TITLE
Fix IPv6

### LINKED ISSUE #2: NameService: unexpected IPv6 record constraint
### Problem

Unexpected NameService IPv6 record constraint (https://github.com/neo-project/non-native-contracts/blob/master/src/NameService/NameService.cs#L476):
```
if (fragments[0] != "2001") return false;
```

### Expected behaviour
```
if (fragments[0] == "2001") return false;
```

### PR DESCRIPTION
**Pull request by erikzhang**
Close #2

### CODE REVIEW

**Comment by erikzhang** [comment]
@AnnaShaleva Please check.

**Review by AnnaShaleva — DISMISSED** [review]

**Comment by roman-khimov** [comment]
If we only want to allow current public (global unicast) addresses (I'm not sure about that) then there is still some room for improvement based on 
https://datatracker.ietf.org/doc/html/rfc5156
https://www.iana.org/assignments/ipv6-address-space/ipv6-address-space.xhtml

(some addresses at 2001 prefix are special)

But at the same time it's already much more usable than what we had before this PR.

**Comment by erikzhang** [comment]
@roman-khimov Fixed.

File: src/NameService/NameService.cs [hunk 1]
```diff
@@ -473,20 +473,59 @@ private static bool CheckIPv6(string ipv6)
...
+                }
+            }
+            ushort number = numbers[0];
+            if (number < 0x2000 || number == 0x2002 || number == 0x3ffe || number > 0x3fff)
+                return false;
+            if (number == 0x2001)
+            {
+                number = numbers[1];
+                if (number < 0x200) return false;
+                if (number == 0xdb8)
```
Comment (roman-khimov) [inline src/NameService/NameService.cs]: Why don't we ban all of `2001:db8::/32`?

File: src/NameService/NameService.cs [hunk 2]
```diff
@@ -473,20 +473,59 @@ private static bool CheckIPv6(string ipv6)
...
+                    numbers[index] = (ushort)StdLib.Atoi(fragment, 16);
+                }
+            }
+            ushort number = numbers[0];
+            if (number < 0x2000 || number == 0x2002 || number == 0x3ffe || number > 0x3fff)
+                return false;
+            if (number == 0x2001)
+            {
+                number = numbers[1];
+                if (number < 0x200) return false;
```
Comment (roman-khimov) [inline src/NameService/NameService.cs]: Seems to be too restrictive. We only need `2001:10::/28`, `2001:20::/28` and `2001::/32` here.

File: src/NameService/NameService.cs (hunk 1 above)
Comment (erikzhang) [reply src/NameService/NameService.cs]: You are right, my bad.

Comment (erikzhang) [reply src/NameService/NameService.cs]: Fixed.

File: src/NameService/NameService.cs (hunk 2 above)
Comment (erikzhang) [reply src/NameService/NameService.cs]: > 2001:0000::/23 is reserved for IETF Protocol Assignments

https://www.iana.org/assignments/ipv6-unicast-address-assignments/ipv6-unicast-address-assignments.xhtml

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**